{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0ce64960-872e-4ebb-8344-b3dbd7348d1c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#import all neccessary libraries\n",
    "import os\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np; np.random.seed(42)\n",
    "from pathlib import Path\n",
    "from io import StringIO \n",
    "import io\n",
    "import os\n",
    "from matplotlib_venn import venn2, venn2_unweighted\n",
    "from matplotlib_venn import venn3, venn3_circles, venn3_unweighted\n",
    "from mpl_toolkits import axes_grid1\n",
    "import matplotlib.ticker as ticker"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "24231358-87dd-4b04-8cda-fb0a08587d01",
   "metadata": {},
   "outputs": [],
   "source": [
    "###Coverage of target region calculation in AS and WGS###\n",
    "\n",
    "AS = pd.read_csv(\"results/AS_sec.target.regions.bed.gz\"\n",
    "                     , sep=\"\\t\",names=[\"chr\",\"start\",\"stop\",\"name\",\"cov\"])\n",
    "WG = pd.read_csv(\"results/WGS_sec.regions.bed.gz\"\n",
    "                     , sep=\"\\t\",names=[\"chr\",\"start\",\"stop\",\"name\",\"cov\"])\n",
    "\n",
    "AS.drop_duplicates(inplace=True)\n",
    "WG.drop_duplicates(inplace=True)\n",
    "\n",
    "#set chromosome size#\n",
    "from collections import OrderedDict\n",
    "s = \"\"\"chr1\t248956422\n",
    "chr2\t242193529\n",
    "chr3\t198295559\n",
    "chr4\t190214555\n",
    "chr5\t181538259\n",
    "chr6\t170805979\n",
    "chr7\t159345973\n",
    "chr8\t145138636\n",
    "chr9\t138394717\n",
    "chr10\t133797422\n",
    "chr11\t135086622\n",
    "chr12\t133275309\n",
    "chr13\t114364328\n",
    "chr14\t107043718\n",
    "chr15\t101991189\n",
    "chr16\t90338345\n",
    "chr17\t83257441\n",
    "chr18\t80373285\n",
    "chr19\t58617616\n",
    "chr20\t64444167\n",
    "chr21\t46709983\n",
    "chr22\t50818468\n",
    "chrX\t156040895\n",
    "chrY\t57227415\"\"\".split(\"\\n\")\n",
    "chr_sizes = OrderedDict(l.split(\"\\t\") for l in s)\n",
    "mpl.style.use('classic')\n",
    "\n",
    "ASdf = AS.sort_values(by=[\"chr\",'start'])\n",
    "WGdf = WG.sort_values(by=[\"chr\",'start'])\n",
    "# Plot configuration\n",
    "mpl.rcParams.update(\n",
    "    {\n",
    "        \"font.size\": 10,\n",
    "        \"figure.facecolor\": \"w\",\n",
    "        \"axes.facecolor\": \"w\",\n",
    "        \"axes.spines.right\": False,\n",
    "        \"axes.spines.top\": False,\n",
    "        \"xtick.top\": False,\n",
    "        \"xtick.bottom\": False,\n",
    "        \"ytick.right\": False,\n",
    "        \"ytick.left\": False,\n",
    "        \"figure.dpi\": 400})\n",
    "lastsize = 0\n",
    "chr_offset=dict()\n",
    "for chr in chr_sizes:\n",
    "    print (chr,lastsize,chr_sizes[chr])\n",
    "    chr_offset[chr]=lastsize\n",
    "    lastsize+=int(chr_sizes[chr])\n",
    "\n",
    "ASdf['chrom']=ASdf['chr'].str.replace('chr','')#.astype(int)\n",
    "\n",
    "ASdf = ASdf.sort_values(by=[\"chrom\",'start'])\n",
    "ASdf['ind2']=ASdf['start'].cumsum()\n",
    "ASdf['ind'] = range(len(ASdf))\n",
    "ASdf[\"chr_offset\"]=ASdf['chr'].map(chr_offset)\n",
    "ASdf['offsetstart']=(ASdf['stop']-ASdf['start'])/2 + ASdf['start'] +ASdf['chr_offset']\n",
    "ASdf_grouped2 = ASdf.groupby(('chrom'))\n",
    "\n",
    "WGdf['chrom']=WGdf['chr'].str.replace('chr','')#.astype(int)\n",
    "\n",
    "WGdf = WGdf.sort_values(by=[\"chrom\",'start'])\n",
    "WGdf['ind2']=WGdf['start'].cumsum()\n",
    "WGdf['ind'] = range(len(WGdf))\n",
    "WGdf[\"chr_offset\"]=WGdf['chr'].map(chr_offset)\n",
    "WGdf['offsetstart']=(WGdf['stop']-WGdf['start'])/2 + WGdf['start'] +WGdf['chr_offset']\n",
    "WGdf_grouped2 = WGdf.groupby(('chrom'))\n",
    "\n",
    "##plot the graph for AS\n",
    "fig = plt.figure(figsize=(20,4))\n",
    "ax = fig.add_subplot(111)\n",
    "colors = ['green']#,'red','green', 'red']\n",
    "x_labels = []\n",
    "x_labels_pos = []\n",
    "maxval= 0\n",
    "for num, (name, group) in enumerate(ASdf_grouped2):\n",
    "    group.plot(kind='scatter', x='offsetstart', y='cov',color=colors[num % len(colors)],alpha=1,linewidth=0,s=10,edgecolors=None,marker=\"o\", ax=ax)\n",
    "    x_labels.append(name)\n",
    "    x_labels_pos.append((group['offsetstart'].iloc[0-1] - (group['offsetstart'].iloc[-1] - group['offsetstart'].iloc[0])/2))\n",
    "    new_maxval=group['offsetstart'].iloc[0-1]\n",
    "    if new_maxval >= maxval:\n",
    "        maxval = new_maxval\n",
    "ax.set_xticks(x_labels_pos)\n",
    "ax.set_xticklabels(x_labels)\n",
    "ax.set_xlim([0, maxval])\n",
    "ax.set_ylim([0, 50])\n",
    "ax.set_xlabel('Chromosome')\n",
    "ax.set_ylabel('Mean Coverage')\n",
    "ax.set_title('Coverage of COSMIC Panel Targets in AS')\n",
    "\n",
    "plt.margins(0.05,0)\n",
    "plt.show()\n",
    "\n",
    "\n",
    "fig.savefig(\"resultsfig/AS_COSMIC_cov\",dpi = 200)\n",
    "\n",
    "##plot the graph for WGS\n",
    "fig = plt.figure(figsize=(20,4))\n",
    "ax = fig.add_subplot(111)\n",
    "colors = ['green']#,'red','green', 'red']\n",
    "x_labels = []\n",
    "x_labels_pos = []\n",
    "for num, (name, group) in enumerate(WGdf_grouped2):\n",
    "    group.plot(kind='scatter', x='offsetstart', y='cov',color=colors[num % len(colors)],alpha=1,linewidth=0,s=10,edgecolors=None,marker=\"o\", ax=ax)\n",
    "    x_labels.append(name)\n",
    "    x_labels_pos.append((group['offsetstart'].iloc[0-1] - (group['offsetstart'].iloc[-1] - group['offsetstart'].iloc[0])/2))\n",
    "    new_maxval=group['offsetstart'].iloc[0-1]\n",
    "    if new_maxval >= maxval:\n",
    "        maxval = new_maxval\n",
    "ax.set_xticks(x_labels_pos)\n",
    "ax.set_xticklabels(x_labels)\n",
    "ax.set_xlim([0, maxval])\n",
    "ax.set_ylim([0, 50])\n",
    "ax.set_xlabel('Chromosome')\n",
    "ax.set_ylabel('Mean Coverage')\n",
    "ax.set_title('Coverage of COSMIC Panel Targets in WGS')\n",
    "\n",
    "plt.margins(0.05,0)\n",
    "plt.show()\n",
    "\n",
    "fig.savefig(\"results/fig/WGS_COSMIC_cov\",dpi = 200)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "702f855c-65cc-4733-b10a-f8c6efec8a77",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Graph for Comparison of AS Target and Non-target region coverage\n",
    "\n",
    "#Use rglob to extract target and non target coverage for AS and for loop to read in data to a dataframe\n",
    "path = Path(\"results/mergedCP/sec_map/AS_sec\")\n",
    "nontarget_paths = path.rglob(\"*_nontarget.regions.bed.gz\")\n",
    "dfsn=[]\n",
    "for file in nontarget_paths: \n",
    "    print(file)\n",
    "    group = file.name.split(\".\")[0]\n",
    "    print(group)\n",
    "    tmp_dfn = pd.read_csv(file, sep=\"\\t\", header=None, names=[\"chr\",\"start\",\"stop\",\"coverage\"])\n",
    "    tmp_dfn[\"gene\"] = \"Nil\"\n",
    "    tmp_dfn[\"group\"] = group\n",
    "    dfsn.append(tmp_dfn)\n",
    "dfn = pd.concat(dfsn)\n",
    "\n",
    "target_path = path.rglob(\"*_target.regions.bed.gz\")\n",
    "dfst=[]\n",
    "for file in target_path: \n",
    "    print(file)\n",
    "    group = file.name.split(\".\")[0]\n",
    "    print(group)\n",
    "    tmp_dft = pd.read_csv(file, sep=\"\\t\", header=None, names=[\"chr\",\"start\",\"stop\",\"gene\",\"coverage\"])\n",
    "    tmp_dft[\"group\"] = group\n",
    "    dfst.append(tmp_dft)\n",
    "dft = pd.concat(dfst)\n",
    "\n",
    "#Concatenate all dataframes\n",
    "cat_df = pd.concat([dfn, dft])\n",
    "\n",
    "#plot the graph\n",
    "fig, ax =plt.subplots(ncols=1, nrows=1, figsize=(13, 10))\n",
    "cat_df[\"Group\"] = cat_df[\"group\"].map(\n",
    "    {\"AS_subset_nontarget\": 'Non-target Region (AS after subset extraction)', \n",
    "     \"AS_total_nontarget\": 'Non-target Region (AS before subset extraction)',\n",
    "     \"AS_subset_target\": 'Target Region (AS after subset extraction)', \n",
    "     \"AS_total_target\": 'Target Region (AS before subset extraction)'})\n",
    "graph = sns.barplot(\n",
    "    data=cat_df[cat_df[\"chr\"].ne(\"chrM\")], \n",
    "    x=\"chr\", \n",
    "    y=\"coverage\", \n",
    "    hue=\"Group\",\n",
    "    ax=ax)\n",
    "plt.legend(loc='upper right', title='Comparison of AS Target and Non-target region coverage') \n",
    "graph.set_xticklabels(graph.get_xticklabels(), rotation=90) #graph.set_xticklabels(rotation = 90)\n",
    "graph.set_xlabel(\"Mean Coverage\") \n",
    "graph.set_ylabel(\"Chromosome\")\n",
    "plt.show\n",
    "\n",
    "#save\n",
    "fig.savefig(\"results/fig/AS_TvsNT_cov\",dpi = 200)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dd9965e6-9b98-43f2-8689-b9a6830316c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Graph for Comparison of WGS Target and Non-target region coverage\n",
    "\n",
    "#Use rglob to extract target and non target coverage for WGS and for loop to read in data to a dataframe\n",
    "path = Path(\"results/mergedCP/sec_map/WGS_sec\")\n",
    "nontarget_paths = path.rglob(\"*_nontarget.regions.bed.gz\")\n",
    "dfsnR=[]\n",
    "for file in nontarget_paths: \n",
    "    print(file)\n",
    "    group = file.name.split(\".\")[0]\n",
    "    print(group)\n",
    "    tmp_dfnR = pd.read_csv(file, sep=\"\\t\", header=None, names=[\"chr\",\"start\",\"stop\",\"coverage\"])\n",
    "    tmp_dfnR[\"gene\"] = \"Nil\"\n",
    "    tmp_dfnR[\"group\"] = group\n",
    "    dfsnR.append(tmp_dfnR)\n",
    "dfnR = pd.concat(dfsnR)\n",
    "\n",
    "target_path = path.rglob(\"*_target.regions.bed.gz\")\n",
    "dfstR=[]\n",
    "for file in target_path: \n",
    "    print(file)\n",
    "    group = file.name.split(\".\")[0]\n",
    "    print(group)\n",
    "    tmp_dftR = pd.read_csv(file, sep=\"\\t\", header=None, names=[\"chr\",\"start\",\"stop\",\"gene\",\"coverage\"])\n",
    "    tmp_dftR[\"group\"] = group\n",
    "    dfstR.append(tmp_dftR)\n",
    "dftR = pd.concat(dfstR)\n",
    "\n",
    "#Concatenate all dataframes\n",
    "catR_df = pd.concat([dfnR, dftR])\n",
    "\n",
    "#Graph for Comparison of AS Target and Non-target region coverage\n",
    "fig, ax =plt.subplots(ncols=1, nrows=1, figsize=(13, 10))\n",
    "catR_df[\"Group\"] = catR_df[\"group\"].map(\n",
    "    {\"WGS_subset_nontarget\": 'Non-target Region (WGS after subset extraction)', \n",
    "     \"WGS_total_nontarget\": 'Non-target Region (WGS before subset extraction)',\n",
    "     \"WGS_subset_target\": 'Target Region (WGS after subset extraction)', \n",
    "     \"WGS_total_target\": 'Target Region (WGS before subset extraction)'})\n",
    "graph = sns.barplot(\n",
    "    data=catR_df[catR_df[\"chr\"].ne(\"chrM\")], \n",
    "    x=\"chr\", \n",
    "    y=\"coverage\", \n",
    "    hue=\"Group\",\n",
    "    ax=ax)\n",
    "plt.legend(loc='upper right', title='Comparison of WGS Target and Non-target region coverage') \n",
    "graph.set_xticklabels(graph.get_xticklabels(), rotation=90) #graph.set_xticklabels(rotation = 90)\n",
    "graph.set_xlabel(\"Mean Coverage\") \n",
    "graph.set_ylabel(\"Chromosome\")\n",
    "plt.show\n",
    "\n",
    "#save the plot\n",
    "fig.savefig(\"results/fig/WGS_TvsNT_cov\",dpi = 200)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f3ae342e-84d2-4ef3-800d-54ab4cc2ddb4",
   "metadata": {},
   "outputs": [],
   "source": [
    "###Graphs to compare basecall HAC vs SAC###\n",
    "\n",
    "##Use panda to read in the data\n",
    "basecall = pd.read_csv(\n",
    "    \"~/mergedCP/basecall/basecall.csv\", sep='\\t',\n",
    "    header=None,\n",
    "    skiprows=1,\n",
    "    names=[\"Metric\", \"Number\", \"Group\"],\n",
    "thousands=\",\")\n",
    "\n",
    "\n",
    "##Barplot for Basecall Accuracy comparison: number of reads\n",
    "sns.set_theme(style=\"whitegrid\")\n",
    "\n",
    "g = sns.catplot(\n",
    "    data=A, \n",
    "    kind=\"bar\",\n",
    "    x=\"Metric\", \n",
    "    y=\"Number\", \n",
    "    hue=\"Group\",\n",
    "    palette=\"dark\", alpha=.8, height=5,aspect=0.8 )\n",
    "\n",
    "g.set_xticklabels([])\n",
    "g.set(xlabel=\"Basecall Accuracy\") \n",
    "g.set(ylabel=\"Number of reads\")\n",
    "g.set(yticks=np.arange(0,275000,25000))\n",
    "g.fig.suptitle(\"Basecall Accuracy comparison for number of reads\",fontsize=11, fontdict={\"weight\": \"bold\"})\n",
    "g.fig.savefig(\"results/fig/basecall1\",dpi = 200)\n",
    "\n",
    "\n",
    "\n",
    "##Barplot for Basecall Accuracy comparison: number of bases\n",
    "B = (basecall[basecall[\"Metric\"] == \"Total bases\"]) \n",
    "sns.set_theme(style=\"whitegrid\")\n",
    "\n",
    "g = sns.catplot(\n",
    "    data=B, \n",
    "    kind=\"bar\",\n",
    "    x=\"Metric\", \n",
    "    y=\"Number\", \n",
    "    hue=\"Group\",\n",
    "   palette=\"dark\", alpha=.8, height=8,aspect=0.4 )\n",
    "g.set(xlabel=\"Basecall Accuracy\") \n",
    "g.set(ylabel=\"Total number of bases\")\n",
    "g.set_xticklabels([])\n",
    "g.set(yticks=np.arange(0,2.5e9,0.05e9))\n",
    "g.legend.set_title(\"\")\n",
    "g.fig.suptitle(\"Basecall Accuracy comparison for total number of bases\",fontsize=11, fontdict={\"weight\": \"bold\"}, va = 'bottom')\n",
    "#save figure\n",
    "g.fig.savefig(\"results/fig/basecall2\",dpi = 200)\n",
    "\n",
    "\n",
    "\n",
    "##Barplot for Basecall Accuracy comparison: Quality of reads\n",
    "#Read in the data with pandas\n",
    "QualityYY = pd.read_csv(\n",
    "    \"~/mergedCP/basecall/quality\", sep='\\t',\n",
    "    header=None,\n",
    "    skiprows=1,\n",
    "    names=[\"Quality_cutoffs\",\t\"HAC\",\t\"SUP\",\t\"HAC%\",\t\"SAC%\",\t\"HAC_Reads\",\"SAC_Reads\"],\n",
    "thousands=\",\")\n",
    "QualityYY\n",
    "\n",
    "#Generat plot\n",
    "A = QualityYY[[\"Quality_cutoffs\",\t\"HAC\",\t\"SUP\"]]\n",
    "B = pd.melt(A, id_vars=['Quality_cutoffs'], value_vars=[\"HAC\",\"SUP\"],var_name='Accuracy')\n",
    "\n",
    "#Plot graph\n",
    "sns.set_theme(style=\"whitegrid\")\n",
    "g = sns.catplot(\n",
    "    data=B, kind=\"bar\",\n",
    "    x='Quality_cutoffs',y = 'value', hue='Accuracy',\n",
    "    ci=\"sd\", palette=\"dark\", alpha=.8, height=5\n",
    ")\n",
    "g.despine(left=True)\n",
    "g.set_axis_labels(\"\", \"Number of reads\")\n",
    "g.legend.set_title(\"\")\n",
    "g.fig.suptitle('Number of reads above quality cutoffs',fontsize=11, fontdict={\"weight\": \"bold\"})\n",
    "#save figure\n",
    "g.fig.savefig(\"results/fig/basecall3\",dpi = 200)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "df7f5552-0f53-407a-b2dd-a3dbcac1dd42",
   "metadata": {},
   "outputs": [],
   "source": [
    "###Venn diagram for HAC and SUP comparison against GIAB truth set ### \n",
    "\n",
    "#Define a function (read_vcf) to read in vcf\n",
    "def read_vcf(path):\n",
    "    with open(path, 'r') as f:\n",
    "        lines = [l for l in f if not l.startswith('##')]\n",
    "    return pd.read_csv(\n",
    "        io.StringIO(''.join(lines)),\n",
    "        dtype={'#CHROM': str, 'POS': int, 'WGS': str, 'ALT': str,}, #'ID': str, 'WGS': str, 'ALT': str,'QUAL': str, 'FILTER': str, 'INFO': str}\n",
    "        sep='\\t',usecols = [\"#CHROM\", \"POS\", \"WGS\", \"ALT\"]\n",
    "    ).rename(columns={'#CHROM': 'CHROM'})\n",
    "# @YoavEtzioni\n",
    "\n",
    "##Read in sup vcfs and intersections with read_vcf function\n",
    "from pathlib import Path\n",
    "path= \"results/mergedCP/pie_sac\"\n",
    "search_path = Path(path)\n",
    "dfs=[]\n",
    "for file in search_path.glob(\"*.vcf\"):\n",
    "    group = file.name.split(\".\")[0]\n",
    "    vcf = read_vcf(file)\n",
    "    vcf[\"group\"] = group\n",
    "    dfs.append(vcf)\n",
    "df = pd.concat(dfs)\n",
    "df.reset_index(inplace=True,drop=True)  \n",
    "\n",
    "#Plot venn diagrams for Super accuracy vs GIAB / GIAB vs AS - HIGH CONFIDENCE REGIO\n",
    "C = df.group.value_counts().Both\n",
    "A = df.group.value_counts().clair3_HC\n",
    "A1= A - C\n",
    "B = df.group.value_counts().GIAB_HC\n",
    "B1 = B - C\n",
    "v = venn2_unweighted(subsets = (A1, B1, C ),\n",
    "      set_labels = ('AS','GIAB'))                   \n",
    "fig = plt.gcf()\n",
    "fig.set_facecolor(\"w\")\n",
    "plt.show\n",
    "plt.title ('SNP called from guppy Super accuracy basecaller output')\n",
    "fig.savefig(\"results/fig/basecall4\",dpi = 200)\n",
    "\n",
    "###Read in hac vcfs and intersections with read_vcf function\n",
    "from pathlib import Path\n",
    "path= \"results/mergedCP/pie2_hac/\"\n",
    "search_path = Path(path)\n",
    "dfs=[]\n",
    "for file in search_path.glob(\"*.vcf\"):\n",
    "    group = file.name.split(\".\")[0]\n",
    "    vcf = read_vcf(file)\n",
    "    vcf[\"group\"] = group\n",
    "    dfs.append(vcf)\n",
    "dfh = pd.concat(dfs)\n",
    "dfh.reset_index(inplace=True,drop=True)  \n",
    "\n",
    "#plot venn diagrams for High accuracy vs GIAB\n",
    "C = dfh.group.value_counts().Both\n",
    "A = dfh.group.value_counts().clair3_HC\n",
    "A1= A - C\n",
    "B = dfh.group.value_counts().GIAB_HC\n",
    "B1 = B - C\n",
    "\n",
    "v=venn2_unweighted(subsets = (A1, B1, C ),\n",
    "      set_labels = ('AS','GIAB'))\n",
    "fig = plt.gcf()\n",
    "fig.set_facecolor(\"w\")    \n",
    "plt.title ('SNP called from guppy High accuracy basecaller output')\n",
    "plt.show\n",
    "fig.savefig(\"results/fig/basecall5\",dpi = 200)\n",
    "\n",
    "\n",
    "#plot venn diagrams GIAB vs WGS - HIGH CONFIDENCE REGION'\n",
    "C = dfall.group.value_counts().GIAB_isect_WGS\n",
    "A = dfall.group.value_counts().GIAB\n",
    "A1= A - C\n",
    "B = dfall.group.value_counts().NA12878\n",
    "B1 = B - C\n",
    "\n",
    "v=venn2_unweighted(subsets = (A1, B1, C ),\n",
    "      set_labels = ('GIAB', 'WGS'))\n",
    "\n",
    "fig = plt.gcf()\n",
    "fig.set_facecolor(\"w\")    \n",
    "plt.title ('GIAB vs WGS - HIGH CONFIDENCE REGION')\n",
    "plt.show\n",
    "\n",
    "fig.savefig(\"results/fig/AS_WGS_HC\",dpi = 200)\n",
    "\n",
    "\n",
    "#plot venn diagrams GIAB vs WGS vs AS - HIGH CONFIDENCE REGION'\n",
    "A = dfall.group.value_counts().GIAB\n",
    "B = dfall.group.value_counts().AS\n",
    "C = dfall.group.value_counts().GIAB_isect_AS\n",
    "D = dfall.group.value_counts().WGS\n",
    "E = dfall.group.value_counts().GIAB_isect_WGS\n",
    "F = dfall.group.value_counts().WGS_isect_AS\n",
    "G = dfall.group.value_counts().All_three_isect\n",
    "C1 = C - G\n",
    "F1 = F - G\n",
    "E1 = E - G\n",
    "A1 = A - (C1 + E1 + G)\n",
    "B1 = B - (C1 + F1 + G)\n",
    "D1 = D - (E1 + F1 + G)\n",
    "\n",
    "v=venn3_unweighted(subsets=(A1, B1, C1, D1, E1 ,F1 ,G), set_labels = ('GIAB', 'AS', 'WGS'))\n",
    "\n",
    "fig = plt.gcf()\n",
    "fig.set_facecolor(\"w\")\n",
    "plt.title ('Comparison of SNP called at High Confidence region in GIAB, AS, WGS')\n",
    "plt.show()\n",
    "fig.savefig(\"results/fig/AS_GIAB_WGS_HC\",dpi = 200)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5e548be9-b273-4f40-8a13-1b6bdb16a6ff",
   "metadata": {},
   "outputs": [],
   "source": [
    "###Graph to plot Distribution of AS coverage in AS only, GIAB only and both###\n",
    "##Read in the table with pandas\n",
    "giab = pd.read_csv(\n",
    "    \"results/mergedCP/variant_calling/gatktables/GIAB.table\", sep='\\t',\n",
    "    header=None,\n",
    "    usecols=[0, 1, 3],\n",
    "    names=[\"chrom\", \"pos\", \"depth\"])\n",
    "giab[\"group\"] = \"GIAB\"\n",
    "Clair3 = pd.read_csv(\n",
    "    \"results/mergedCP/variant_calling/gatktables/AS.table\", sep='\\t',\n",
    "    skiprows=1,\n",
    "    header=None,\n",
    "    names=[\"chrom\", \"pos\", \"depth\"])\n",
    "Clair3[\"group\"] = \"AS\"\n",
    "Both = pd.read_csv(\n",
    "    \"results/mergedCP/variant_calling/gatktables/Both.table\", sep='\\t',\n",
    "    skiprows=1,\n",
    "    header=None,\n",
    "    names=[\"chrom\", \"pos\", \"depth\"])\n",
    "Both[\"group\"] = \"BOTH\"\n",
    "#concatenate the table\n",
    "dfHC_2 = pd.concat([giab, Clair3, Both])\n",
    "\n",
    "#Plot the graph\n",
    "fig, ax =plt.subplots(ncols=1, nrows=1, figsize=(8, 6))\n",
    "\n",
    "ax = sns.violinplot(\n",
    "    data = dfHC_2,\n",
    "    x=\"group\", \n",
    "    y=\"depth\")\n",
    "ax.set_title(\"Distribution of coverage\")\n",
    "ax.set_ylabel(\"Depth\")\n",
    "ax.set_xlabel(\"\")\n",
    "plt.show()\n",
    "\n",
    "fig.savefig(\"results/fig/AS_GIAB_COV\",dpi = 200)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "256032ba-e995-40f9-bc60-9c6778f5f868",
   "metadata": {},
   "outputs": [],
   "source": [
    "WGS_meth = pd.read_csv(\n",
    "    \"results/whole_genome/5mC_NA12878_bedmethyl\", sep='\\t',\n",
    "    header=None,\n",
    "    names=[\"chrom\", \"start\", \"end\", \"name\", \"score\", \"strand\", \"tstart\", \"tend\", \"color\", \"coverage\", \"freq\", \"ref_G\", \"sample_G\", \"qual\"])\n",
    "AS_meth = pd.read_csv(\n",
    "    \"results/methylation_mcp/modbam2bed/5mC_ml32merged_bedmethyl\", sep='\\t',\n",
    "    header=None,\n",
    "    names=[\"chrom\", \"start\", \"end\", \"name\", \"score\", \"strand\", \"tstart\", \"tend\", \"color\", \"coverage\", \"freq\", \"ref_G\", \"sample_G\", \"qual\"])\n",
    "meth_merged = pd.merge(WGS_meth, AS_meth, how=\"inner\", on=[\"chrom\",\"start\",\"end\"], suffixes = (\"_WGS\",\"_AS\"))\n",
    "methyl\n",
    "\n",
    "\n",
    "POLD1 = methyl[methyl['chrom'].eq('chr19') & methyl['start'].between(50384340,50418010)]\n",
    "FREQ_POLD1 = POLD1[[\"WGS_meth\",\"AS_meth\"]]\n",
    "matrix = FREQ_POLD1.corr()\n",
    "print(matrix)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
